### PR TITLE
Scheduler: Remove LoggingClient global variable

### LINKED ISSUE #2035: Scheduler: refactor to remove LoggingClient global variable
Refactor the above-identified service to remove the above-identified global variable (which is defined in the service's init.go).

This work has already been completed in the system management agent.  Use it as a reference.  See https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/router.go and https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/init.go.

You will need to update route creation to explicitly pass the individual required dependencies to the controller functions.  This will bubble down through the implementation; that is, you'll need to pass those dependencies down the call chain to where they're currently referencing the global variable.  Do not pass the dependency injection container into any controller functions.

You can also create any long-lived dependencies in the service's bootstrap and use the dependency injection container to hold them.  This was done in the system management agent; look there for an example.

Add me as the sole reviewer of your PR.

Note this is Geneva work and should not be merged into master prior to Fuji branching.  As there are a series of these issues for this service (each refactoring away a global variable), assign all of this service's related issues to you and use the branch created for this PR as the starting point for your next PR to avoid extensive conflicts.

### PR DESCRIPTION
**Pull request by brandonforster**
Fixes #2035

Signed-off-by: Brandon Forster <brandonforster@gmail.com>

### CODE REVIEW

File: internal/support/scheduler/utils.go [hunk 1]
```diff
@@ -104,11 +106,12 @@ func parseInt64(value string) int64 {
...
 }
 
 // NewClient
-func NewSchedulerQueueClient() *QueueClient {
-	queueClient := &QueueClient{}
+func NewSchedulerQueueClient(loggingClient logger.LoggingClient) *QueueClient {
+	queueClient := &QueueClient{loggingClient}
 	currentQueueClient = queueClient // Set the singleton
 	return queueClient
 }
```
Comment (michaelestrin) [inline internal/support/scheduler/utils.go]: Acknowledge you didn't write this code.

It's a mess.  Can you improve it?  Create a separate GitHub issue.  See individual comments.

File: internal/support/scheduler/utils.go [hunk 2]
```diff
@@ -104,11 +106,12 @@ func parseInt64(value string) int64 {
 // Scheduler Queue Client
 var currentQueueClient *QueueClient // Singleton used so that queueClient can use it to de-reference readings
```
Comment (michaelestrin) [inline internal/support/scheduler/utils.go]: (for new GitHub issue) My IDE indicates this global isn't referenced.  Should probably be removed.

File: internal/support/scheduler/utils.go [hunk 3]
```diff
@@ -104,11 +106,12 @@ func parseInt64(value string) int64 {
...
 type QueueClient struct {
+	loggingClient logger.LoggingClient
 }
```
Comment (michaelestrin) [inline internal/support/scheduler/utils.go]: (for new GitHub issue) QueueClient receiver should be moved to schedule.go where it's methods reside.

File: internal/support/scheduler/utils.go (hunk 1 above)
Comment (michaelestrin) [inline internal/support/scheduler/utils.go]: (for new GitHub issue) QueueClient factory method should be moved to schedule.go where it's implementation lives.

File: internal/support/scheduler/rest_interval_test.go [hunk 4]
```diff
@@ -403,24 +423,24 @@ func createMockIntervalLoaderAddNameInUse() interfaces.DBClient {
...
 	myMock.On("IntervalActionsByIntervalName", intervalForAdd.Name).Return([]contract.IntervalAction{}, nil)
 	myMock.On("UpdateInterval", intervalForAdd).Return(nil)
 	return &myMock
 }
 
 func validateInterval(interval *contract.Interval) {
 	b, _ := json.Marshal(interval)
-	interval.UnmarshalJSON(b)
+	_ = interval.UnmarshalJSON(b)
 }
```
[the remark this answers is not in the document]
Comment (brandonforster) [reply internal/support/scheduler/rest_interval_test.go]: The purpose of this is to set the unexported field `isValidated` which can only be done by marshalling and unmarshalling.

[the remark this answers is not in the document]
Comment (michaelestrin) [reply internal/support/scheduler/rest_interval_test.go]: Makes sense.  Wish it wasn't so.  Appreciate the explanation -- definitely isn't self-evident.

[the remark this answers is not in the document]
Comment (brandonforster) [reply internal/support/scheduler/rest_interval_test.go]: I went ahead and added a comment to note this for the next time it comes up 👍

File: internal/support/scheduler/schedule.go [hunk 5]
```diff
@@ -90,33 +93,33 @@ func (qc *QueueClient) Connect() (string, error) {
...
 	mutex.Lock()
 	defer mutex.Unlock()
 
 	intervalContext, exists := intervalIdToContextMap[intervalId]
 	if !exists {
-		logMsg := fmt.Sprintf("scheduler could not find a interval context with interval id : %s", intervalId)
-		LoggingClient.Info(logMsg)
-		return contract.Interval{}, errors.New(logMsg)
+		return contract.Interval{},
+			errors.New(fmt.Sprintf("scheduler could not find a interval context with interval id : %s", intervalId))
```
Comment (michaelestrin) [inline internal/support/scheduler/schedule.go]: Nitpick: `errors.New(fmt.Sprintf(...))` == `fmt.Errorf`

Commented once; occurs elsewhere.

**Comment by brandonforster** [comment]
recheck

**Pull request merged**